receiver/githubreceiver/trace_event_handling.go

```diff
@@ -130,3 +163,201 @@ func (gtr *githubTracesReceiver) createRootSpan(
+	event *github.WorkflowJobEvent,
+	traceID pcommon.TraceID,
+	parentSpanID pcommon.SpanID,
+) error {
+	steps := event.GetWorkflowJob().Steps
+	unique := newUniqueSteps(steps)
+	var errors error
+	for i, step := range steps {
+		name := unique[i]
+		err := gtr.createStepSpan(resourceSpans, traceID, parentSpanID, event, step, name)
+		if err != nil {
+			errors = multierr.Append(errors, err)
+		}
+	}
+	return errors
+}
+
+// newUniqueSteps creates a new slice of step names from the provided GitHub
+// event steps. Each step name, if duplicated, is appended with `_n` where n is
+// the numbered occurrence.
+func newUniqueSteps(steps []*github.TaskStep) []string {
+	if len(steps) == 0 {
+		return nil
+	}
+
+	results := make([]string, len(steps))
+
+	count := make(map[string]int, len(steps))
+	for _, step := range steps {
+		count[step.GetName()]++
+	}
+
+	occurrences := make(map[string]int, len(steps))
+	for i, step := range steps {
+		name := step.GetName()
+		if count[name] == 1 {
+			results[i] = name
+			continue
+		}
+
+		occurrences[name]++
+		if occurrences[name] == 1 {
+			results[i] = name
+		} else {
+			results[i] = fmt.Sprintf("%s-%d", name, occurrences[name]-1)
+		}
+	}
+
+	return results
+}
+
+// createStepSpan creates a span with a deterministic spandID for the provided
+// step.
+func (gtr *githubTracesReceiver) createStepSpan(
+	resourceSpans ptrace.ResourceSpans,
+	traceID pcommon.TraceID,
+	parentSpanID pcommon.SpanID,
+	event *github.WorkflowJobEvent,
+	step *github.TaskStep,
+	name string,
+) error {
+	scopeSpans := resourceSpans.ScopeSpans().AppendEmpty()
+	span := scopeSpans.Spans().AppendEmpty()
+	span.SetName(name)
+	span.SetKind(ptrace.SpanKindServer)
+	span.SetTraceID(traceID)
+	span.SetParentSpanID(parentSpanID)
+
+	runID := event.GetWorkflowJob().GetRunID()
+	runAttempt := int(event.GetWorkflowJob().GetRunAttempt())
+	jobName := event.GetWorkflowJob().GetName()
+	stepName := step.GetName()
+	number := int(step.GetNumber())
+	spanID, err := newStepSpanID(runID, runAttempt, jobName, stepName, number)
+	if err != nil {
+		return fmt.Errorf("failed to generate step span ID: %w", err)
+	}
+
+	span.SetSpanID(spanID)
+
+	attrs := span.Attributes()
+	attrs.PutStr(semconv.AttributeCicdPipelineTaskName, name)
+	attrs.PutStr(AttributeCICDPipelineTaskRunStatus, step.GetStatus())
+	span.SetStartTimestamp(pcommon.NewTimestampFromTime(step.GetStartedAt().Time))
+	span.SetEndTimestamp(pcommon.NewTimestampFromTime(step.GetCompletedAt().Time))
+
+	switch strings.ToLower(step.GetConclusion()) {
+	case "success":
+		attrs.PutStr(AttributeCICDPipelineTaskRunStatus, AttributeCICDPipelineTaskRunStatusSuccess)
+		span.Status().SetCode(ptrace.StatusCodeOk)
+	case "failure":
+		attrs.PutStr(AttributeCICDPipelineTaskRunStatus, AttributeCICDPipelineTaskRunStatusFailure)
+		span.Status().SetCode(ptrace.StatusCodeError)
+	case "skipped":
+		attrs.PutStr(AttributeCICDPipelineTaskRunStatus, AttributeCICDPipelineTaskRunStatusFailure)
```

Comment:
```suggestion
		attrs.PutStr(AttributeCICDPipelineTaskRunStatus, AttributeCICDPipelineTaskRunStatusSkip)
```
_Unless I'm missing something_